pkg/controller/configurationpolicy/configurationpolicy_controller.go

```diff
@@ -490,11 +490,15 @@ func createInformStatus(mustNotHave bool, numCompliant int, numNonCompliant int,
 		//noncompliant; mustnothave and objects exist
 		update = createMustNotHaveStatus(kind, nonCompliantObjects, namespaced, plc, indx, compliant)
 	}
-	if !mustNotHave && numCompliant > 0 {
+	if !mustNotHave && numCompliant > 0 && numNonCompliant == 0 {
 		//compliant; musthave and objects exist
 		compliant = true
 		update = createMustHaveStatus("", kind, compliantObjects, namespaced, plc, indx, compliant)
 	}
+	if !mustNotHave && numNonCompliant > 0 {
+		//noncompliant; musthave and some objects do not exist
+		update = createMustHaveStatus("", kind, nonCompliantObjects, namespaced, plc, indx, compliant)
+	}
```

Comment:
If `numCompliant == 0` and `numNonCompliant == 1`, then both the condition on line 484 and the condition on line 498 are true. So, the second match will overwrite the first. They are both "noncompliant" statuses, so it's not terrible, but it feels like a bad thing.

I think it should be refactored so that all the conditions are mutually exclusive, and the behavior is more obvious.